pagination/pager.go

```diff
@@ -113,3 +124,106 @@ func (p Pager) EachPage(handler func(Page) (bool, error)) error {
 		}
 	}
 }
+
+// AllPages returns all the pages from a `List` operation in a single page,
+// allowing the user to retrieve all the pages at once.
+func (p Pager) AllPages() (Page, error) {
+	// Having a value of `nil` for `p.PageType` will cause a run-time error.
+	if p.PageType == nil {
+		return nil, fmt.Errorf("Pager field PageType must be set to successfully call pagination.AllPages method.")
+	}
+	// pagesSlice holds all the pages until they get converted into as Page Body.
+	var pagesSlice []interface{}
+	// body will contain the final concatenated Page body.
+	var body reflect.Value
+
+	// Grab a test page to ascertain the page body type.
+	testPage, err := p.fetchNextPage(p.initialURL)
+	if err != nil {
+		return nil, err
+	}
+
+	// Switch on the page body type. Recognized types are `map[string]interface{}`,
+	// `[]byte`, and `[]interface{}`.
+	switch testPage.GetBody().(type) {
+	case map[string]interface{}:
+		// key is the map key for the page body if the body type is `map[string]interface{}`.
+		var key string
+		// Iterate over the pages to concatenate the bodies.
+		err := p.EachPage(func(page Page) (bool, error) {
+			b := page.GetBody().(map[string]interface{})
+			for k := range b {
+				// If it's a linked page, we don't want the `links`, we want the other one.
+				if k != "links" {
```

Comment:
This is basically trying to extract all the elements out of a nested object without knowing the resource key, right? The only thing we know is that we want to ignore links. But most OpenStack link structures are often in the form of `servers_links` not `links`. So maybe we should use `strings.Contains` here instead?